src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -199,42 +200,43 @@ public async Task<Manifests> DeserializeExistingManifestsAndUpdate(List<string>
-                out List<WingetCreateCore.Models.Installer.Installer> unmatchedInstallers,
-                out List<WingetCreateCore.Models.Installer.Installer> multipleMatchedInstallers,
-                out List<PackageParser.DetectedArch> detectedArchOfInstallers))
-            {
-                DisplayMismatchedArchitectures(detectedArchOfInstallers);
+            List<PackageParser.DetectedArch> detectedArchOfInstallers;
 
-                if (installerMismatch)
-                {
-                    Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
+            try
+            {
+                PackageParser.UpdateInstallerNodesAsync(
+                    installerManifest,
+                    this.InstallerUrls,
+                    packageFiles,
+                    out detectedArchOfInstallers);
+            }
+            catch (InvalidOperationException)
+            {
+                Logger.ErrorLocalized(nameof(Resources.InstallerCountMustMatch_Error));
+                return null;
+            }
+            catch (Exceptions.ParsePackageException parsePackageException)
+            {
+                var parseFailedInstallerUrls = parsePackageException.ParseFailedInstallerUrls;
+                parseFailedInstallerUrls.ForEach(i => Logger.ErrorLocalized(nameof(Resources.PackageParsing_Error), i));
+                return null;
+            }
+            catch (Exceptions.InstallerMatchException installerMatchException)
+            {
+                var multipleMatchedInstallers = installerMatchException.MultipleMatchedInstallers;
+                var unmatchedInstallers = installerMatchException.UnmatchedInstallers;
 
-                    if (unmatchedInstallers.Any())
-                    {
-                        Logger.ErrorLocalized(nameof(Resources.InstallerMatchFailedError_Message));
-                        Console.WriteLine();
-                        unmatchedInstallers.ForEach(i => Logger.ErrorLocalized(nameof(Resources.InstallerDetectedFromUrl_Message), i.Architecture, i.InstallerType, i.InstallerUrl));
-                    }
+                Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
+                Logger.ErrorLocalized(nameof(Resources.InstallersFailedToMatch_Message));
 
-                    if (multipleMatchedInstallers.Any())
-                    {
-                        Logger.ErrorLocalized(nameof(Resources.MultipleMatchingInstallerNodes_Error));
-                        Console.WriteLine();
-                        multipleMatchedInstallers.ForEach(i =>
-                        {
-                            Logger.ErrorLocalized(nameof(Resources.InstallerDetectedFromUrl_Message), i.Architecture, i.InstallerType, i.InstallerUrl);
-                            Console.WriteLine();
-                        });
-                    }
+                if (multipleMatchedInstallers.Any())
+                {

```

Comment:
This is redundant, the ForEach won't do anything if there's !Any

Reply:
Removed Any check